common/djangoapps/student/views.py

```diff
@@ -708,6 +711,11 @@ def dashboard(request):
         'course_programs': course_programs,
     }
 
+    referer = request.META.get('HTTP_REFERER')
+    if referer:
+        if referer.find("register") >= 0 or referer.find("oauth") > 0:
+            return redirect("/welcome-unactivated/")
+
```

Comment:
Also, don't assume `/welcome-unactivated` exists. The `edx-platform` is supposed to be generic, not focused on Cognitive Class stack.

I would use `settings.MKTG_URLS['WELCOME_UNACTIVATED']` as the conditional for redirecting.